src/Shared/ValueTaskExtensions/ValueTaskExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static Task GetAsTask(this in ValueTask<FlushResult> valueTask)
             if (valueTask.IsCompletedSuccessfully)
             {
                 // Signal consumption to the IValueTaskSource
-                valueTask.GetAwaiter().GetResult();
+                _ = valueTask.Result;
```

Comment:
Revert this change.

Reply:
This was my main point in this PR. I thought using `Result` is better than `GetAwaiter().GetResult()` when `IsCompletedSuccessfully` is `true`. Could you please explain why this change is not good?

Also I found couple of similar scenario which are using `Result`
https://github.com/aspnet/AspNetCore/blob/b23ea5b6683b08e8d168ccf49e0fc8515077ee2e/src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvoker.cs#L431
https://github.com/aspnet/AspNetCore/blob/5e575a3e64254932c1fd4937041a4e7426afcde4/src/Http/Routing/src/EndpointRoutingMiddleware.cs#L65
https://github.com/aspnet/AspNetCore/blob/0a1d68b8f375eb979c5722b8bedd57496749b1f8/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageLoaderMatcherPolicy.cs#L79
https://github.com/aspnet/AspNetCore/blob/caa910ceeba5f2b2c02c47a23ead0ca31caea6f0/src/Servers/Kestrel/Core/src/Internal/Infrastructure/PipeWriterHelpers/TimingPipeFlusher.cs#L60
https://github.com/aspnet/AspNetCore/blob/94edcbf65bc68aeebc97be8347577304059a01e0/src/Mvc/Mvc.ViewFeatures/src/Controller.cs#L369

Is there any difference between theses and this case?

Reply:
/cc @pranavkm